### PR TITLE
Add support for remote models on sagemaker

### PR DESCRIPTION
**Pull request by JonnySpruce**
Adds support for calling models running on SageMaker to calculate logits

### CODE REVIEW

**Review by jstrong-scottlogic — APPROVED** [review]
LGTM, the doc string is good as well.
